documents/description.md

```diff
@@ -2,8 +2,105 @@
+
+I will probably want to either write my own parser or use something more sophisticated than the PackRat parser used in class since I
+will want to be able to run extensive tests on my code to provide the most interesting error messages and feedback to the user.
+Although this maybe possible with the PackRat parser, I believe there are easier ways to do it -- I will need to investigate this
+however.
+
+###### What kinds of things might go wrong in a program in this domain (e.g., syntax errors, compile-time errors, run-time errors)?
+Since programs are essentially "compiling" and being interpreted by the game, all of these problems are possible and need to be 
+detected. Of these errors, however, the most interesting is the runtime error. This is because the server will need to run at 
+something like 30fps regardless of how much code the user provides. Thus, a runtime error that needs to be accounted for is how long
+a snippet of code takes to run. If a user provides code that will take "too long to run," even if it would eventually terminate, the
+game needs a way to detect this and report it as some type of runtime error.
+
+###### How might you design your language to prevent such errors or to clearly communicate the results of errors to the user?
+Although I could certainly design the language to not resemble C, that is in many ways the point of the game (I do hope to eventually 
+add other languages like a psudo-LISP later). I want the game to help users feel comfortable with common programming language idioms 
+(i.e. ending lines with semicolons, using parenthesis to call functions, returning values, etc). Thus, since the syntax may not
+be as clear as it _could_ be, the idea is to provide in game voiced feedback about the user's code. Thus, an error in the users 
+program will be reported to the user by a character saying something like, "I think you forgot a semicolon here!"
+
+
+
 ## Example computations
+
+###### Describe some example computations in your DSL. These computations should describe what happens when a specific program in your DSL executes. Note that you shouldn't describe the syntax of the program. Rather you should describe some canonical examples that help someone else understand the kinds of things that your DSL will eventually be able to do.
+Although the point of this game is to be extensible and sandbox-like, here are some scenarios I have already imagined:
+   * Program a ship to have a space "dogfight" with other spaceships
+   * Program a fleet to have a coordinated space battle with enemy spaceships
+   * Program a ship to naviagate an astroid belt without colliding with rocks or debris
+   * Program a ship to naviagate a supply route to provide fringe space colonies with supplies
+   * Program a ship to locate and gather minerals deposits from asteroids
+   * Program a flagship to construct additional ships from a selection of parts
+
```

Comment:
There's a lot of good detail in this description. In future milestones (and in the final writeup), it would be good to edit the writeups for clarity and cohesion (so you don't have to repeat yourself as much).